ui/packages/catalog-ui-search/src/main/webapp/component/query-template-sharing/query-template-sharing.view.js

```diff
@@ -0,0 +1,209 @@
+            }
+        ));
+
+        this.listenTo(this.action.currentView.model, 'change:value', this.updateAction);
+    }
+});
+
+let RoleSharingEditor = Marionette.CollectionView.extend({
+    childView: SharingByRoleView
+});
+
+module.exports = Marionette.LayoutView.extend({
+    template: template,
+    tagName: CustomElements.register('template-sharing'),
+    modelEvents: {
+        reset: 'render',
+        sync: 'cleanup'
+    },
+    regions: {
+        byEmail: '.template-sharing-by-email',
+        byRole: '.template-sharing-by-role'
+    },
+    events: {
+        'click .save': 'save',
+        'click .reset': 'render' // resetting can be done by re-rendering and flushing the state
+    },
+    serializeData: function () {
+        return {
+            link: window.location.href + '/' + this.model.get('id')
+        }
+    },
+    getSharingByEmail: function () {
+        return this.options.permissions.accessIndividuals != undefined ? 
+            this.options.permissions.accessIndividuals
+            .map(function (email) {
+                return { value: email }
+            }) : [];
+    },
+    getSharingByRole: function () {
+        let view = this;
+
+        let roles = this.options.permissions.accessGroups != undefined ? 
+            this.options.permissions.accessGroups
+            .map(function (group) {
+                return group;
+            }) : [];
+
+        return user.get('user').get('roles').map(function (role) {
+            return {
+                attribute: 'role',
+                action: (roles.indexOf(role) === -1) ? 'none' : 'access',
+                value: role
+            };
+        });
+    },
+    onRender: function () {
+        this.collection = new Backbone.Collection(this.getSharingByRole());
+        this.emailCollection = new Backbone.Collection(this.getSharingByEmail());
+
+        this.byEmail.show(new EmailSharingEditor({
+            collection: this.emailCollection
+        }));
+
+        this.byRole.show(new RoleSharingEditor({
+            collection: this.collection
+        }));
+    },
+    save: function () {
+
+        let view = this;
+
+        Loading.beginLoading(view);
+
+        let emailList = this.emailCollection.map(function(email) {
+            return email.get('value')
+        });
+
+        let roleList = this.collection.chain()
+            .filter(function (role) {
+                return role.get('action') !== 'none';
+            }).map(function (role) {
+                return role.get('value');
+            }).value();
+
+        let templatePerms = {
+                'security.access-individuals': emailList,
+                'security.access-groups': roleList 
+        }
+
+        this.updateSharingPermissions(templatePerms)
+    },
+    updateSharingPermissions: function(templatePerms) {
+        let _this = this;
+        $.ajax({
+            url: "/search/catalog/internal/sharing/".concat(_this.options.modelId),
```

Comment:
✏️ i think you can use `context: this` in the ajax call and avoid having to set it outside